### PR TITLE
Add w3d model loading test

### PR DESCRIPTION
**Pull request by feliwir**

### CODE REVIEW

File: tests/test_w3d_load.cpp [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        idx++;
+    }
+}
+
+TEST(w3d_model, validate_chunk_loader)
+{
+    auto filepath = Utf8String(TESTDATA_PATH) + "/models/cube.w3d";
+
+    BufferedFileClass file(filepath);
+    ASSERT_TRUE(file.Open(1));
```
Comment (xezon) [inline tests/test_w3d_load.cpp]: FM_READ

File: tests/test_w3d_load.cpp [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    ChunkLoadClass cload(&file);
+    EXPECT_TRUE(cload.Open_Chunk());
+    EXPECT_EQ(cload.Cur_Chunk_ID(), W3D_CHUNK_MESH);
+
+    // Traverse through the model to validate chunks
+    load_mesh(cload);
+
+    cload.Close_Chunk();
+}
```
Comment (xezon) [inline tests/test_w3d_load.cpp]: eol

File: tests/test_w3d_load.cpp [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    auto filepath = Utf8String(TESTDATA_PATH) + "/models/cube.w3d";
+
+    BufferedFileClass file(filepath);
+    ASSERT_TRUE(file.Open(1));
+    ChunkLoadClass cload(&file);
+    EXPECT_TRUE(cload.Open_Chunk());
+    EXPECT_EQ(cload.Cur_Chunk_ID(), W3D_CHUNK_MESH);
+
+    // Traverse through the model to validate chunks
+    load_mesh(cload);
```
Comment (xezon) [inline tests/test_w3d_load.cpp]: Why is this in separate function?

File: tests/test_w3d_load.cpp [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+TEST(w3d_model, load_model)
+{
+    // We need to have an instance of this
+    W3DAssetManager assetmngr;
+    auto filepath = Utf8String(TESTDATA_PATH) + "/models/cube.w3d";
+
+    BufferedFileClass file(filepath);
+    ASSERT_TRUE(file.Open(1));
```
Comment (xezon) [inline tests/test_w3d_load.cpp]: FM_READ

**Review by xezon — APPROVED** [review]

**Comment by feliwir** [comment]
Fixed :)

**Review by xezon — APPROVED** [review]

**Pull request merged**